infrastructure/persistence/dispatcher_repository.go

```diff
@@ -1,28 +1,46 @@
 package persistence
 
 import (
+	"bytes"
 	"config-manager/domain"
+	"context"
+	"encoding/json"
 	"fmt"
+	"net/http"
 )
 
-type DispatcherRepository struct {
-	DispatcherURL string
-	RunStatusURL  string
-	PlaybookURL   string
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
 }
 
-// Placeholder
-func (r *DispatcherRepository) Dispatch(clientID string) (*domain.DispatcherResponse, error) {
-	fmt.Println("Sending request to playbook dispatcher for client: ", clientID)
-	res := &domain.DispatcherResponse{
-		Code:  200,
-		RunID: "id-from-playbook-dispatcher",
-	}
-	return res, nil
+type DispatcherClient struct {
+	DispatcherHost string
+	DispatcherPSK  string
+	Client         HTTPClient
 }
 
-// Placeholder
-func (r *DispatcherRepository) GetStatus(label string) ([]domain.DispatcherRun, error) {
-	fmt.Println("Getting status for hosts using label: ", label)
-	return nil, nil
+func (r *DispatcherClient) Dispatch(
+	ctx context.Context,
+	inputs []domain.DispatcherInput,
+) ([]domain.DispatcherResponse, error) {
+	fmt.Println("Sending request to playbook dispatcher")
+
+	reqBody, err := json.Marshal(inputs)
+	if err != nil {
+		panic(err)
```

Comment:
Watch out for these panics here.  This might make the entire process fall over.  That might be what you want for right now since we are pretty early on in the integration/testing proess but watch out for these as we get closer to production.